### PR TITLE
Feat/ota 3883/image Add new endpoint to receive image manifests

### PR DESCRIPTION
**Pull request by simao**
A manifest is associated with a commit and represents information
about the manifests.xml used to build the image associated with that
commit.

The uploaded manifest is stored as raw json in treehub's database but
some information is parsed and sent to the bus.

Needs https://github.com/advancedtelematic/libats/pull/131

### CODE REVIEW

**Comment by simao** [comment]
Tests failing until libats is merged

File: src/main/scala/com/advancedtelematic/treehub/db/Schema.scala [hunk 1]
```diff
@@ -59,4 +61,16 @@ object Schema {
   }
 
   protected[db] val refs = TableQuery[RefTable]
+
+  case class ManifestTable(tag: Tag) extends Table[CommitManifest](tag, "manifest") {
+    def namespace = column[Namespace]("namespace")
+    def commit: Rep[ObjectId] = column[ObjectId]("object_id")
+    def content = column[Json]("content")
+
+    def pk = primaryKey("pk_ref", (namespace, commit)) // TODO: Check this
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/treehub/db/Schema.scala]: Still need to check something??

File: src/main/scala/com/advancedtelematic/treehub/http/ManifestResource.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+object ManifestParser {
+  case class ManifestInfo(releaseBranch: String, metaUpdaterVersion: String)
+
+  implicit val manifestDecoder: Decoder[ManifestInfo] = Decoder.instance { acursor =>
+    val cursor = acursor.downField("manifest")
+
+    val releaseBranchVersionsEither = for {
+      releaseBranch <- cursor.downField("default").downField("@revision").as[String]
+      metaUpdaterVersion <- cursor.downField("project").as[List[Map[String, String]]]
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/treehub/http/ManifestResource.scala]:
```suggestion
      metaUpdaterVersions <- cursor.downField("project").as[List[Map[String, String]]]
```

File: src/main/scala/com/advancedtelematic/treehub/http/ManifestResource.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    } yield releaseBranch -> metaUpdaterVersion
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/treehub/http/ManifestResource.scala]:
```suggestion
    } yield releaseBranch -> metaUpdaterVersions
```

File: build.sbt [hunk 4]
```diff
@@ -27,10 +27,10 @@ lazy val root = (project in file("."))
...
   .settings(testOptions in IntegrationTest := Seq(Tests.Filter(itFilter)))
   .settings(Seq(libraryDependencies ++= {
-    val akkaV = "2.5.23"
-    val akkaHttpV = "10.1.8"
-    val scalaTestV = "3.0.0"
-    val libatsV = "0.3.0-38-g6acedb6"
+    val akkaV = "2.5.25"
+    val akkaHttpV = "10.1.10"
+    val scalaTestV = "3.0.8"
+    val libatsV = "0.3.0-49-g9dcb9ac-SNAPSHOT"
```
Comment (jochenschneider) [inline build.sbt]: Please don't forget to update this.

**Pull request merged**